src/App.tsx

```diff
@@ -1,26 +1,83 @@
-import { useSelector } from 'react-redux';
+import { useDispatch, useSelector } from 'react-redux';
 
 import './App.scss';
-import Start from './components/Start';
-import { Finish } from './components/Finish';
+import React, { useCallback, useEffect, useState } from 'react';
+import lodash from 'lodash';
+import TodoList from './components/TodoList';
 
-import { isLoading, getMessage } from './store';
+import { getingTodos } from './store';
+import Pagination from './components/Pagination/Pagination';
+import { Loader } from './components/Loader';
+import { fetchTodos } from './store/action-creators/todos';
+import { TodoFilter } from './components/TodoFilter';
+import { TodoModal } from './components/TodoModal';
+import { PaginationActionTypes } from './store/reducers/paginationReducer';
 
 const App = () => {
-  const loading = useSelector(isLoading);
-  const message = useSelector(getMessage) || 'Ready!';
+  const [filterBy, setFilterBy] = useState('all');
+  const [checkedTodo, setCheckedTodo] = useState(true);
+  const [appliedQuery, setAppliedQuery] = useState('');
+  const {
+    todos, error, loading, todo,
+  } = useSelector(getingTodos);
+  const dispatch = useDispatch();
+
+  const handleUserChange = (event: React.ChangeEvent<HTMLSelectElement>) => {
+    setFilterBy(event.target.value);
+  };
+
+  const applyQuery = useCallback(
+    lodash.debounce(setAppliedQuery, 1000), [],
+  );
+
+  const resetInputValue = () => {
+    setAppliedQuery('');
+  };
+
+  useEffect(() => {
+    dispatch(fetchTodos());
+  }, []);
+
+  useEffect(() => {
+    dispatch({
+      type: PaginationActionTypes.UPDATE_TOTAL_AMOUNT,
+      payload: todos.length,
+    });
+  }, [todos]);
+
+  const openModalHandler = () => {
+    setCheckedTodo(true);
+  };
+
+  const closeModalHandler = () => {
+    setCheckedTodo(false);
+  };
 
   return (
     <div className="App">
-      <h1>Redux list of todos</h1>
-      <h2>{loading ? 'Loading...' : message}</h2>
-
-      <Start title="Start loading" />
-      <Finish title="Succeed loading" message="Loaded successfully!" />
-      <Finish
-        title="Fail loading"
-        message="An error occurred when loading data."
-      />
+      <h1 className="title">Redux list of todos</h1>
+      {loading && <Loader />}
+      {loading && error && (
+        <h2>{error}</h2>
```

Comment:
when there is an error after loading will it be shown ?
when there is an error will it be shown with a loader together?